packages/core/bootstrap/src/lib/ws/reducer.ts

```diff
@@ -0,0 +1,87 @@
+import hash from 'object-hash'
+import { AdapterRequest } from '@chainlink/types'
+import { combineReducers, createReducer } from '@reduxjs/toolkit'
+import * as actions from './actions'
+import { WSConfig } from './types'
+import { getHashOpts } from '../util'
+import { logger } from '../external-adapter'
+
+export const getSubsId = (subscriptionMsg: any): string => hash(subscriptionMsg, getHashOpts())
+export interface ConnectionsState {
+  /** Map of all connections by key */
+  active: {
+    [key: string]: WSConfig
+  }
+  /** Number of active connections */
+  total: number
+}
+
+const initConnectionsState: ConnectionsState = { active: {}, total: 0 }
+
+export const connectionsReducer = createReducer<ConnectionsState>(
+  initConnectionsState,
+  (builder) => {
+    builder.addCase(actions.connected, (state, action) => {
+      logger.info('WS: New connection')
```

Comment:
Please remove all logging for reducers, as they should be side-effect-free, and move them to epics where events are triggering.